src/kili/services/helpers.py

```diff
@@ -91,6 +86,18 @@ def infer_ids_from_external_ids(
     return id_map
 
 
+def _build_id_map(auth, asset_external_ids, project_id):
+    assets = AssetQuery(auth.client)(
+        AssetWhere(project_id),
+        ["id", "externalId"],
+        QueryOptions(disable_tqdm=True),
+    )
+    id_map: Dict[str, str] = {}
+    for asset in (asset for asset in assets if asset["externalId"] in set(asset_external_ids)):
```

Comment:
you should put the `set(asset_external_ids)` outside of the generator, because a new set will be computed for each asset in assets

Reply:
no it should not be recomputed, see:
```
In [1]: a = 5

In [2]: for b in range(a):
   ...:     print(a)
   ...:     a+=1
   ...:
5
6
7
8
9

```

Reply:
I'm not 100% sure honestly.

This is not the same "in" in the code and in your example.

The "in" in for `for b in range(a)` is an "iterator" in.

The "in" in ` ... in set(...)` is a membership test. Everytime it's executed, the set() will be evaluated again:

```python
class MyClass:
    def __init__(self):
        self.a = [1, 2, 3]

    def __call__(self):
        print("class called")
        return self.a

a = MyClass()
for nb in (nb for nb in [1, 2, 3] if nb in set(a())):
    print("for loop: ", nb)
```

prints:

```python
class called
for loop: 1
class called
for loop: 2
class called
for loop: 3
```

Reply:
oh you're right, my mistake